sdk/storage/storage-blob/rollup.base.config.js

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import nodeResolve from "rollup-plugin-node-resolve";
+import multiEntry from "rollup-plugin-multi-entry";
+import cjs from "rollup-plugin-commonjs";
+import json from "rollup-plugin-json";
+import replace from "rollup-plugin-replace";
+import { uglify } from "rollup-plugin-uglify";
+import sourcemaps from "rollup-plugin-sourcemaps";
+import shim from "rollup-plugin-shim";
+// import visualizer from "rollup-plugin-visualizer";
+
+const version = require("./package.json").version;
+const banner = [
+  "/*!",
+  ` * Azure Storage SDK for JavaScript - Blob, ${version}`,
+  " * Copyright (c) Microsoft and contributors. All rights reserved.",
+  " */"
+].join("\n");
+
+const pkg = require("./package.json");
+const depNames = Object.keys(pkg.dependencies);
+const production = process.env.NODE_ENV === "production";
+
+export function nodeConfig(test = false) {
+  const externalNodeBuiltins = [
+    "@azure/ms-rest-js",
+    "crypto",
+    "fs",
+    "events",
+    "os",
+    "stream"
+  ];
+  const baseConfig = {
+    input: "dist-esm/src/index.js",
+    external: depNames.concat(externalNodeBuiltins),
+    output: {
+      file: "dist/index.js",
+      format: "cjs",
+      sourcemap: true
+    },
+    plugins: [
+      sourcemaps(),
+      replace({
+        delimiters: ["", ""],
+        values: {
+          // replace dynamic checks with if (true) since this is for node only.
+          // Allows rollup's dead code elimination to be more aggressive.
+          "if (isNode)": "if (true)"
+        }
+      }),
+      nodeResolve({ preferBuiltins: true }),
+      cjs(),
+      json()
+    ]
+  };
+
+  if (test) {
+    // entry point is every test file
+    baseConfig.input = [
+      "dist-esm/test/*.spec.js",
+      "dist-esm/test/node/*.spec.js"
+    ];
+    baseConfig.plugins.unshift(multiEntry({ exports: false }));
+
+    // different output file
+    baseConfig.output.file = "dist-test/index.node.js";
+
+    // mark assert as external
+    baseConfig.external.push("assert", "fs", "path");
+
+    baseConfig.onwarn = warning => {
```

Comment:
Interesting, for 10.1.0-preview, we have similar roll up config for Node.js test cases. However, doesn't have met this "THIS_IS_UNDEFINED" warning. Can you check why previous version doesn't have this issue?

See https://github.com/Azure/azure-storage-js/blob/v10.1.0-preview-blob/blob/rollup.test.config.js

Reply:
It's possible this ignore was added because it was needed in other projects and isn't needed here. Harsha can confirm.

Reply:
We do see "THIS_IS_UNDEFINED" warning which I could only get rid of using custom onwarn handlers.
Also, I've verified v10.1.0-preview branch, I need to look deeper on why it didn't throw those warnings previously.
I'm guessing this might be from version updates in the dependencies[not really sure].

Reply:
**Update -** 

- The following line made the difference.
`node.context = "null"`
https://github.com/Azure/azure-storage-js/blob/v10.1.0-preview-blob/blob/rollup.test.config.js#L9

- By commenting that line, I was able to reproduce the `(!) 'this' has been rewritten to 'undefined'` warnings with the code in `v10.1.0-preview-blob` branch(tag).

- With the code in the current PR, and by leveraging this line of code(`context = "null"`) instead of onwarn handler, I am still able to get rid of the warnings.

From the documentation https://rollupjs.org/guide/en#danger-zone, it is not clear to me on what `context = "null"` does.

Reply:
Thanks @HarshaNalluru ! Here is the description about "THIS NOT DEFINED" in rollup doc. `options.context` is a workaround to override default value of `this`. @bterlson previously added `node.context = "null"` in v10.1.0-preview. I believe this approach should work. 

Error: "this is undefined"
In a JavaScript module, this is undefined at the top level (i.e., outside functions). Because of that, Rollup will rewrite any this references to undefined so that the resulting behaviour matches what will happen when modules are natively supported.

There are occasional valid reasons for this to mean something else. If you're getting errors in your bundle, you can use options.context and options.moduleContext to change this behaviour.

Reply:
Removed onwarn handlers and updated with `context = "null"` 
2d2c526